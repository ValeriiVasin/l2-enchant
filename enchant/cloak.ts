import { EnchantMap, Item, Enchant } from '../types';

const chances = [70, 65, 58, 42, 32, 32, 48, 28, 28, 28];
const items = [
  Item.Cloak,
  Item.Cloak_1,
  Item.Cloak_2,
  Item.Cloak_3,
  Item.Cloak_4,
  Item.Cloak_5,
  Item.Cloak_6,
  Item.Cloak_7,
  Item.Cloak_8,
  Item.Cloak_9,
  Item.Cloak_10,
];

export const cloakEnchantMap: EnchantMap = new Map([
  ...items.slice(0, -1).map((item, index): [Item, Enchant] => [
    item,
    {
      item,
      required: Item.CloakEnchant,
      successRate: chances[index],
      success: items[index + 1],
      fail: items[Math.max(index - 1, 0)],
    },
  ]),
]);
