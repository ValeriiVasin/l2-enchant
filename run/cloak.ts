import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.Cloak,
  [...itemsRange(Item.Cloak_1, Item.Cloak_10), ...itemsRange(Item.LegendaryCloak_10, Item.LegendaryCloak_15)],
  100_000,
);
