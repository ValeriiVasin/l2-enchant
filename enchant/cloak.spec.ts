import { cloakEnchantMap } from './cloak';

it('enchant map: cloak', () => {
  expect(cloakEnchantMap).toMatchSnapshot();
});
