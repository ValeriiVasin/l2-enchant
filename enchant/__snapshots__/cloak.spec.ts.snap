// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: cloak 1`] = `
Map {
  "Плащ" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Плащ +1",
      },
      Object {
        "chance": 30,
        "item": "Плащ",
      },
    ],
  },
  "Плащ +1" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +1",
    "results": Array [
      Object {
        "chance": 65,
        "item": "Плащ +2",
      },
      Object {
        "chance": 35,
        "item": "Плащ",
      },
    ],
  },
  "Плащ +2" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +2",
    "results": Array [
      Object {
        "chance": 58,
        "item": "Плащ +3",
      },
      Object {
        "chance": 42,
        "item": "Плащ +1",
      },
    ],
  },
  "Плащ +3" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +3",
    "results": Array [
      Object {
        "chance": 42,
        "item": "Плащ +4",
      },
      Object {
        "chance": 58,
        "item": "Плащ +2",
      },
    ],
  },
  "Плащ +4" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +4",
    "results": Array [
      Object {
        "chance": 32,
        "item": "Плащ +5",
      },
      Object {
        "chance": 68,
        "item": "Плащ +3",
      },
    ],
  },
  "Плащ +5" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +5",
    "results": Array [
      Object {
        "chance": 32,
        "item": "Плащ +6",
      },
      Object {
        "chance": 68,
        "item": "Плащ +4",
      },
    ],
  },
  "Плащ +6" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +6",
    "results": Array [
      Object {
        "chance": 48,
        "item": "Плащ +7",
      },
      Object {
        "chance": 52,
        "item": "Плащ +5",
      },
    ],
  },
  "Плащ +7" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +7",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Плащ +8",
      },
      Object {
        "chance": 72,
        "item": "Плащ +6",
      },
    ],
  },
  "Плащ +8" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +8",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Плащ +9",
      },
      Object {
        "chance": 72,
        "item": "Плащ +7",
      },
    ],
  },
  "Плащ +9" => Object {
    "cost": "Свиток: Модифицировать Плащ",
    "item": "Плащ +9",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Плащ +10",
      },
      Object {
        "chance": 72,
        "item": "Плащ +8",
      },
    ],
  },
  "Плащ +10" => Object {
    "cost": "Легендарный Свиток",
    "item": "Плащ +10",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Легендарный Плащ +10",
      },
      Object {
        "chance": 95,
        "item": "Плащ +10",
      },
    ],
  },
  "Легендарный Плащ +10" => Object {
    "cost": "Стабильный Свиток: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +10",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Легендарный Плащ +11",
      },
      Object {
        "chance": 72,
        "item": "Легендарный Плащ +10",
      },
    ],
  },
  "Легендарный Плащ +11" => Object {
    "cost": "Стабильный Свиток: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +11",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Легендарный Плащ +12",
      },
      Object {
        "chance": 72,
        "item": "Легендарный Плащ +11",
      },
    ],
  },
  "Легендарный Плащ +12" => Object {
    "cost": "Стабильный Свиток: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +12",
    "results": Array [
      Object {
        "chance": 48,
        "item": "Легендарный Плащ +13",
      },
      Object {
        "chance": 52,
        "item": "Легендарный Плащ +12",
      },
    ],
  },
  "Легендарный Плащ +13" => Object {
    "cost": "Стабильный Свиток: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +13",
    "results": Array [
      Object {
        "chance": 48,
        "item": "Легендарный Плащ +14",
      },
      Object {
        "chance": 52,
        "item": "Легендарный Плащ +13",
      },
    ],
  },
  "Легендарный Плащ +14" => Object {
    "cost": "Стабильный Свиток: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +14",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Легендарный Плащ +15",
      },
      Object {
        "chance": 80,
        "item": "Легендарный Плащ +14",
      },
    ],
  },
  "Легендарный Плащ +15" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +15",
    "results": Array [
      Object {
        "chance": 7,
        "item": "Легендарный Плащ +16",
      },
      Object {
        "chance": 93,
        "item": "Легендарный Плащ +15",
      },
    ],
  },
  "Легендарный Плащ +16" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +16",
    "results": Array [
      Object {
        "chance": 7,
        "item": "Легендарный Плащ +17",
      },
      Object {
        "chance": 93,
        "item": "Легендарный Плащ +16",
      },
    ],
  },
  "Легендарный Плащ +17" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +17",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Легендарный Плащ +18",
      },
      Object {
        "chance": 95,
        "item": "Легендарный Плащ +17",
      },
    ],
  },
  "Легендарный Плащ +18" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +18",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Легендарный Плащ +19",
      },
      Object {
        "chance": 95,
        "item": "Легендарный Плащ +18",
      },
    ],
  },
  "Легендарный Плащ +19" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Легендарный Плащ",
    "item": "Легендарный Плащ +19",
    "results": Array [
      Object {
        "chance": 3,
        "item": "Легендарный Плащ +20",
      },
      Object {
        "chance": 97,
        "item": "Легендарный Плащ +19",
      },
    ],
  },
}
`;
